[feat-login-normal] 로그인 관련 기능

gopythor opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
 - 로그인 코드 심플화
 - 토큰 / 보안 등 최소화
 - 멤버 소프트 삭제 기능 추가


**TO-BE**
 - OAuth2 및 쿠키 로그인 기능 추가

### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [x] 테스트 코드
- [x] API 테스트 

- 테스트 코드 Mockito 사용
- API 테스트는 Swagger로 진행

gopythor committed:
일반 login 기능 간소화

71mC40H0 commented on src/main/java/com/zerototen/savegame/config/WebConfig.java:
```diff
@@ -1,8 +1,11 @@
 package com.zerototen.savegame.config;
 
 import com.zerototen.savegame.util.StringToLocalDateConverter;
+import org.springframework.context.annotation.Bean;
```
이거랑 밑에 추가로 import 하신 부분들은 지금 이 파일에 쓰이지 않는 것 같은데 넣으신 이유가 궁금합니다.

gopythor replied:
제가 정리를 못했나봅니다. 삭제처리 하겠습니다.

71mC40H0 commented on src/main/java/com/zerototen/savegame/controller/MemberController.java:
```diff
@@ -31,6 +32,11 @@ public ResponseEntity<SaveDto> register(@RequestBody MemberDto.SaveDto request)
         return memberService.register(request);
     }
 
+    @DeleteMapping("delete")
```
@DeleteMapping 만 그냥 쓰셔도 될 것 같습니다.

71mC40H0 commented on src/main/java/com/zerototen/savegame/service/MemberService.java:
```diff
@@ -76,46 +58,60 @@ public ResponseEntity<MemberDto.SaveDto> register(MemberDto.SaveDto request) {
+    }
+
+    // 로그인 시 메일 및 비밀번호 검증
     private void LOGIN_VALIDATE(MemberDto.LoginDto request) {
```
메소드명은 camelCase로 수정해주시면 될 것 같습니다.

gopythor replied:
옙 수정 하겠습니다

71mC40H0 commented on src/main/java/com/zerototen/savegame/service/MemberService.java:
```diff
@@ -76,46 +58,60 @@ public ResponseEntity<MemberDto.SaveDto> register(MemberDto.SaveDto request) {
     }
 
+    // 회원 가입 시 검증
     private void REGISTER_VALIDATION(MemberDto.SaveDto request) {
```
이 부분도 메소드명을 camelCase로 수정해주시면 될 것 같습니다.

gopythor replied:
옙 수정하겠습니다.

71mC40H0 commented on src/main/java/com/zerototen/savegame/util/Aes256Util.java:
```diff
@@ -31,15 +28,14 @@ public static String decrypt(String cipherText) {
             byte[] decodedBytes = Base64.decodeBase64(cipherText);
             byte[] decrypted = cipher.doFinal(decodedBytes);
-            return new String(decrypted, StandardCharsets.UTF_8);
+            return new String(decrypted,StandardCharsets.UTF_8);
```
이 부분은 왜 수정되었을까요?? 코드 정렬을 한 번 해주시면 될 것 같습니다.

gopythor replied:
옙 정렬하겠습니다.

71mC40H0 commented on src/test/java/com/zerototen/savegame/service/MemberServiceTest.java:
```diff
@@ -1,66 +1,158 @@
 
-        // when
-        memberService.register(member);
+        SaveDto newMember = SaveDto.builder()
```
객체 생성 관련 부분은 밑에 private 메소드로 빼서 사용하시면 좋을 것 같습니다.

gopythor replied:
무슨 말씀인지 잘 이해가 안되는데 한번 검색/공부하고 private 메소드 빼도록 하겠습니다.

71mC40H0 replied:
제가 작성한 테스트 코드 보시면 맨 밑에 get으로된 메소드들이 있을거에요. 그것처럼 따로 빼서 처리하시면 될 것 같습니다.

71mC40H0 reviewed (commented):
몇 가지 리뷰 드렸습니다. 고생하셨습니다.

gopythor committed:
refactoring
1. 불필요 import 정리
2. delete에 / 등 추가
3. 변수 2개(Login Validate / Register Validate) camelCase로 수정
4. 코드 정렬
5. 테스트코드 private 생성자 구현
